azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -461,14 +461,15 @@ private boolean isDeployToSlot() {
         return slotSetting != null && StringUtils.isNotEmpty(slotSetting.getName());
     }
 
-    private ApplicationInsightsComponent getOrCreateApplicationInsights(boolean enableCreation) throws AzureAuthFailureException {
-        final AzureTokenCredentials credentials = getAzureTokenWrapper();
+    private ApplicationInsightsComponent getOrCreateApplicationInsights(boolean enableCreation) throws AzureAuthFailureException, AzureExecutionException {
+        final AzureCredentialWrapper credentials = getAzureCredentialWrapper();
         if (credentials == null) {
             Log.warn(APPLICATION_INSIGHTS_NOT_SUPPORTED);
             return null;
         }
         final String subscriptionId = getAzureClient().subscriptionId();
-        final ApplicationInsightsManager applicationInsightsManager = new ApplicationInsightsManager(credentials, subscriptionId, getUserAgent());
+        final ApplicationInsightsManager applicationInsightsManager = new ApplicationInsightsManager(credentials.getAzureTokenCredentials(),
+                subscriptionId, getUserAgent());
```

Comment:
will we use other properties of `AzureTokenCredentionWrapper` in Mojo except for `AzureTokenCredential`